var/spack/repos/builtin/packages/superlu/package.py

```diff
@@ -4,49 +4,50 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 
-class Superlu(Package):
+class Superlu(CMakePackage):
     """SuperLU is a general purpose library for the direct solution of large,
     sparse, nonsymmetric systems of linear equations on high performance
     machines. SuperLU is designed for sequential machines."""
 
     homepage = "http://crd-legacy.lbl.gov/~xiaoye/SuperLU/#superlu"
-    url      = "http://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_5.2.1.tar.gz"
+    url      = "https://github.com/xiaoyeli/superlu/archive/refs/tags/v5.2.2.tar.gz"
 
+    version('5.2.2', sha256='470334a72ba637578e34057f46948495e601a5988a602604f5576367e606a28c')
     version('5.2.1', sha256='28fb66d6107ee66248d5cf508c79de03d0621852a0ddeba7301801d3d859f463')
-    version('4.3', sha256='169920322eb9b9c6a334674231479d04df72440257c17870aaa0139d74416781')
+    version('4.3', sha256='169920322eb9b9c6a334674231479d04df72440257c17870aaa0139d74416781',
+            url='https://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_4.3.tar.gz')
+    version('4.2', sha256='5a06e19bf5a597405dfeea39fe92aa8c5dd41da73c72c7187755a75f581efb28',
+            url='https://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_4.2.tar.gz')
```

Comment:
Should we deprecate these 4.X versions?

Reply:
Probably. I put them in there because a couple codebases I've been working with depend on these particular versions due to API changes (which ended up being rather minor). Shall I